### PR TITLE
[WIP] Get rid of save button

### PR DESCRIPTION
**Pull request by meganyin13**
### Summary 
Got rid of save button and extra space formatting for save. Added save functionality to onMouseLeave.

This pull request is the first step towards implementing feature Foo

- [x] implemented X
- [ ] fixed Y

### Breaking Changes  

- Database schema change (anything that changes Firestore collection structure)
- Other change that could cause problems (Detailed in notes)

### CODE REVIEW

File: frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx [hunk 1]
```diff
@@ -94,11 +94,11 @@ function TaskEditor(
...
-  const onMouseLeave = (): void => {
-    if (onBlur) {
-      onBlur();
-    }
-  };
+  // const onMouseLeave = (): void => {
+  //   if (onBlur) {
+  //     onBlur();
+  //   }
+  // };
```
Comment (SamChou19815) [inline frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx]: Do not leave commented code. We have version control.

**Comment by mt-xing** [comment]
Does tabbing out of the dialog box work as expected?

**Comment by meganyin13** [comment]
@mt-xing I have pushed changes to add `onSaveClicked` to `onBlur` but when adding a new subtask when you type one letter it jumps out, saves that, and then you have to click in to finish editing. Changes are saved though when tabbing out.

**Comment by mt-xing** [comment]
It's a little janky, what with needing to move the mouse away before it actually saves, rather than just checking off the task when clicked as a user might expect. We can talk about this more tomorrow. Good work so far.

**Comment by SamChou19815** [comment]
Looks like more work needs to be done on the `TaskEditor` component. I will still lose focus when I press enter in the last subtask input box.

**Pull request closed without merging**